astroid/objects.py

```diff
@@ -314,11 +314,13 @@ def qname(self):
 class Property(scoped_nodes.FunctionDef):
     """Class representing a Python property"""
 
+    @decorators.deprecate_arguments(doc="Use the postinit arg 'doc_node' instead")
     def __init__(
         self, function, name=None, doc=None, lineno=None, col_offset=None, parent=None
     ):
         self.function = function
-        super().__init__(name, doc, lineno, col_offset, parent)
+        super().__init__(name, lineno=lineno, col_offset=col_offset, parent=parent)
+        self._doc = doc
```

Comment:
Without this we raise the `DeprecationWarning` from the `__init__` of `FunctionDef`. This avoid this.